testsuite/classlibrary/TestSCDoc.sc

```diff
@@ -0,0 +1,25 @@
+TestSCDoc : UnitTest {
+
+    test_helpSourceDirs_excludedExtensions {
+        var result;
+        var extensionPaths = [thisProcess.platform.systemExtensionDir, thisProcess.platform.userExtensionDir];
+        var prevSetting = LanguageConfig.excludeDefaultPaths;
+        LanguageConfig.excludeDefaultPaths = true;
+        SCDoc.indexAllDocuments;
```

Comment:
this is a very expensive call -- if there's no quicker way to check this behavior right now, i would recommend refactoring the SCDoc code so it's possible.

Reply:
thanks! I just had another go at it, adding helpSourceDirs = nil to *initClass and calling it inside tests to refresh helpSourceDirs. If this is too hacky I could add a class method just to set helpSourceDirs to nil... or any other idea is welcome :)

Reply:
yeah, that seems a bit hacky. i wouldn't expect `initClass` to do this, we now require an extra comment to explain why that line can't be removed from `initClass` even though it is more or less unnecessary.

there is some code in indexAllDocuments that could be touched up at the same time as we're fixing this:

https://github.com/supercollider/supercollider/blob/develop/SCClassLibrary/SCDoc/SCDoc.sc#L465-L466

i would suggest one of two options:

1. add `prRescanHelpSourceDirs` such that you can replace those two lines with `this.prRescanHelpSourceDirs.do {|dir|`
2. add `prResetHelpSourceDirs` such that you can replace the first line with `this.prResetHelpSourceDirs`

and then use whichever of these you implement in your test code.

i'm not completely sold on adding this tiny function, but the lazy-evaluation semantics of `helpSourceDirs` kind of call for it. and, i think it would be better to call a clearly-named private function in test code than a confusingly-named public one.